src/meta/src/stream/source_manager.rs

```diff
@@ -715,7 +719,7 @@ impl SourceManager {
         let current_splits_ref = splits.clone();
         let source_id = source.id;
 
-        let connector_properties = extract_prop_from_source(&source)?;
+        let connector_properties = extract_prop_from_source(&source, false)?;
```

Comment:
When will we have unknown fields from the view of the newer version? If we neither do any migration nor warn users here, does it mean that some information is lost?

Reply:
> the view of the newer version

I don't get it. What do you mean

Reply:
In other words, in which case an unknown field can be safely ignored?

Reply:
When can't it be ignored? Can you give an example?

Reply:
Since this is the current behavior, are you discussing some future problems? 🤔

Reply:
For instance, a field is accidentally renamed without providing an alias.

Reply:
I'm still not 100% sure about your intention. Are you thinking about
- hard deny without allowing unknown options
- or use protobuf to avoid developer mistakes? If so, I think `with_options.yaml` can somehow function as a compatibility test?

Reply:
Yeah, both of them.

Reply:
I now think users can tolerate non-fatal errors more than being forced to rebuild the whole stuff. So I want to go ahead with this now and do that in future if we are sure that's needed.